datafusion/core/src/physical_optimizer/join_selection.rs

```diff
@@ -176,7 +176,7 @@ fn swap_join_projection(
 /// This function swaps the inputs of the given join operator.
 /// This function is public so other downstream projects can use it
 /// to construct `HashJoinExec` with right side as the build side.
-pub fn swap_hash_join(
+pub(crate) fn swap_hash_join(
```

Comment:
I believe this is actually a downgrade of visibility -- it used to be `pub` but now is `pub(crate)` meaning that comet can't use it: https://github.com/apache/datafusion/issues/13898

I will make a PR to fix this

Reply:
Oops, sorry for that. I guess we hastily thought `pub(crate)` would be sufficient but in Comet's case they need the full `pub`.

Reply:
No worries -- it was easily fixed